module/documents/actor.js

```diff
@@ -361,100 +385,32 @@ export default class Actor5e extends Actor {
 
   /**
-   * Convert a bonus value to a simple integer for displaying on the sheet.
-   * @param {number|string|null} bonus  Actor's bonus value.
-   * @param {object} data               Actor data to use for replacing @ strings.
-   * @returns {number}                  Simplified bonus as an integer.
-   * @protected
-   */
-  _simplifyBonus(bonus, data) {
-    if ( !bonus ) return 0;
-    if ( Number.isNumeric(bonus) ) return Number(bonus);
-    try {
-      const roll = new Roll(bonus, data);
-      if ( !roll.isDeterministic ) return 0;
-      roll.evaluate({ async: false });
-      return roll.total;
-    } catch(error) {
-      console.error(error);
-      return 0;
-    }
-  }
-
-  /* -------------------------------------------- */
-
-  /**
-   * Initialize derived AC fields for Active Effects to target.
-   * Mutates the system.attributes.ac object.
-   * @private
-   */
-  _prepareBaseArmorClass() {
-    const ac = this.system.attributes.ac;
-    ac.armor = 10;
-    ac.shield = ac.bonus = ac.cover = 0;
-    this.armor = null;
-    this.shield = null;
-  }
-
-  /* -------------------------------------------- */
-
-  /**
-   * Calculate the initiative bonus to display on a character sheet.
-   * Mutates the value of the system.attributes.init object.
-   * @param {number} globalCheckBonus  The simplified global ability check bonus for this actor
-   * @param {object} bonusData         Actor data to use for replacing formula variables in bonuses
-   */
-  _computeInitiativeModifier(globalCheckBonus, bonusData) {
-    const flags = this.flags.dnd5e || {};
-    const init = this.system.attributes.init;
-    const dex = this.system.abilities.dex;
-
-    // Initiative modifiers
-    const joat = flags.jackOfAllTrades;
-    const athlete = flags.remarkableAthlete;
-    const dexCheckBonus = this._simplifyBonus(dex?.bonuses?.check, bonusData);
-
-    // Compute initiative modifier
-    init.mod = dex?.mod ?? 0;
-    init.prof = new Proficiency(this.system.attributes.prof, (joat || athlete) ? 0.5 : 0, !athlete);
-    init.value = init.value ?? 0;
-    init.bonus = init.value + (flags.initiativeAlert ? 5 : 0);
-    init.total = init.mod + init.bonus + dexCheckBonus + globalCheckBonus;
-    if ( Number.isNumeric(init.prof.term) ) init.total += init.prof.flat;
-  }
-
-  /* -------------------------------------------- */
-
-  /**
-   * Derive any values that have been scaled by the Advancement system.
-   * Populates the system.scale object.
+   * Prepare data related to the spell-casting capabilities of the Actor.
+   * Mutates the value of the system.spells object.
    * @private
    */
-  _computeScaleValues() {
-    const scale = this.system.scale ??= {};
-    for ( const [identifier, cls] of Object.entries(this.classes) ) {
-      scale[identifier] = cls.scaleValues;
-      if ( cls.subclass ) scale[cls.subclass.identifier] = cls.subclass.scaleValues;
-    }
+  _computeSpellcastingProgression() {
```

Comment:
A few comments here:
1. If something was a public method, we should add a deprecation warning for it and shift it to the bottom of section of the file, make sure to tag it as `@deprecated` for IDE hinting.
2. If something was not previously public like `_computeSpellcastingProgression`, I think we can (even should) remove it directly without any sort of backwards-compatible warning.

Reply:
While normally I'd agree, I think in the instance of these preparation methods we should still leave the deprecation warnings in. Because of the lack of hooks in 5e up to this point, the only way to enhance any of this logic has been to wrap these private methods, so I think it is preferable to avoid immediately bricking a whole bunch of modules.